src/ok.rs

```diff
@@ -59,6 +247,88 @@ pub fn cmd(gen: Option<SevGeneration>, quiet: bool) -> Result<()> {
     }
 }
 
-fn run_test() -> bool {
-    true
+fn add_sev_es_test(tests: &mut Vec<Test>) {
+    let sev_es_test = Test {
+        name: "Encrypted State (SEV-ES)",
+        run: Box::new(|| {
+            let res = unsafe { x86_64::__cpuid(0x8000001f) };
+
+            let stat = if (res.eax & 0x1 << 3) != 0 {
+                TestState::Pass
+            } else {
+                TestState::Fail
+            };
+
+            TestResult {
+                name: "Encrypted State (SEV-ES)",
+                stat,
+                mesg: None,
+            }
+        }),
+        sub: vec![],
+    };
+
+    tests[0].sub[2].sub.push(sev_es_test);
```

Comment:
This looks a little too magical. Could the tests have a bitmask or an enum or something that identifies whether it should run or not? Then when running the tests, we can pass a bitmask/enum/flag/whatever that corresponds to the set of tests that should be considered active. If it's not active, just mark the test and its sub-tests as skip.

Reply:
Sure, I've added some generation-specific bitmasks to have "conditional execution" of the tests in my latest commit.